src/test/scala/mesosphere/marathon/api/v2/ModelValidationTest.scala

```diff
@@ -74,6 +80,34 @@ class ModelValidationTest
       v.message == "Requested service port 3200 conflicts with a service port in app /app2") should be(true)
   }
 
+  test("Multiple errors within one field of a validator should be grouped into one array") {
+    val empty = ImportantTitle("")
+
+    validate(empty) match {
+      case Success => fail()
+      case f: Failure =>
+        val errors = (Json.toJson(f) \ "details").as[Seq[JsObject]]
+        errors should have size 1
+        (errors.head \ "path").as[String] should be("name")
+        (errors.head \ "errors").as[Seq[String]] should have size 2
+    }
+  }
+
+  test("Validators should not produce 'value' string at the end of description.") {
+    val group = Group("/test/group1".toPath, Set(
+      AppDefinition("/test/group1/valid".toPath, cmd = Some("foo")),
+      AppDefinition("/test/group1/invalid".toPath)
+    ))
+
+    validate(group) match {
+      case Success => fail()
+      case f: Failure =>
+        val errors = (Json.toJson(f) \ "details").as[Seq[JsObject]]
+        errors should have size 1
+        (errors.head \ "path").as[String] should be("apps[1]")
```

Comment:
The `self` part is described in the upper comment. Regarding `groups[0]` - there is only one group here, containing two apps. The validation is done on that group object, so there is no list of groups in it.